### ISSUE TITLE
2.0 doesn't respect target="_blank" attr

### ISSUE THREAD

**Issue by hogart** [vuejs/vue-router#827]
Following code doesn't open new tab/window:

``` jade
router-link("v-bind:to"="{name: 'play', params: {ifid: ifid}}", target="_blank") Play (new tab)
```

I'm upgrading my app from 1.x where appropriate equivalent of this code worked. Here's repro pen: http://codepen.io/hogart/pen/ZpPGXY?editors=1010

**Comment by fnlctrl** [comment]
It's no longer supported, please use `<a>` tags instead.

**Comment by hogart** [comment]
>  use <a> tags

Srsly? Well, this lib gone south really fast:(

**Comment by hogart** [comment]
Manually creating links is error prone and can't be really considered solution — аlas, router instance doesn't provide method for reverse url resolving.

I understand that this is quite rare use case, but this is obvious regression and as such should be at least mentioned in upgrade guide.

**Comment by fnlctrl** [comment]
> аlas, router instance doesn't provide method for reverse url resolving.

It has.. though it's not documented (since it's very unlikely that someone needs it):
`router.match()`, you can put whatever in `router-link`'s `to` prop to it, e.g. `router.match({name:'foo'})`
and to get the url, use `router.match({name:'foo'}).fullPath`

> I understand that this is quite rare use case, but this is obvious regression and as such should be at least mentioned in upgrade guide.

Sorry about that, just created a PR to fix this: https://github.com/vuejs/vuejs.org/pull/539/files

**Comment by hogart** [comment]
Yeah, I already dug out this method, although it's not full equivalent of what's happening in `router-link` render function. Can I rely upon it in a sense it get's documented some time and won't disappear?

**Comment by fnlctrl** [comment]
>  although it's not full equivalent of what's happening in router-link render function.

Yeah, it doesn't handle `base`.. https://github.com/vuejs/vue-router/blob/dev/src/components/link.js#L30-L33

>  Can I rely upon it in a sense it get's documented some time and won't disappear?

It's unlikely to be documented, and I can't guarantee its consistence... Please use at your own risk.

**Comment by posva** [comment]
@fnlctrl Couldn't we implement this with `window.open` and provide a `target` prop?

**Comment by fnlctrl** [comment]
@posva Evan fixed this for 0.7 (https://github.com/vuejs/vue-router/commit/c955be39a0929a1139fc167fe4881296a9e0cef4), but I don't know if he intentionally removed this in 2.0. It would be easy to support though.

**Comment by posva** [comment]
@fnlctrl I think we should 😄 
Let's talk about this on slack

**Comment by hogart** [comment]
@fnlctrl while target="_blank" is indeed corner case, reverse url resolve can be useful in other cases. Imagine we're building isomorphic app and we need to email user some link. Or we doing something like google drive or bit.ly and we need to provide links in text form to copy them.

**Comment by fnlctrl** [comment]
Well for the "target=_blank" issue I've created a PR: https://github.com/vuejs/vue-router/pull/830
@hogart As for the reverse url resolve, would you mind opening a new issue for discussion?

**Comment by LinusBorg** [comment]
> Srsly? Well, this lib gone south really fast:(

Sidenote: Is that kind of behaviour necessary?

**Comment by yyx990803** [comment]
fixed by merging #830
